salt/returners/local_cache.py

```diff
@@ -255,6 +255,8 @@ def save_minions(jid, minions, syndic_id=None):
         minions_path = os.path.join(jid_dir, MINIONS_P)
 
     try:
+        if not os.path.exists(jid_dir):
+            os.makedirs(jid_dir)
```

Comment:
@cachedout this needs another try/except block around the makedirs, since this happens in multiple processes at once this can be a race, the try/except should catch it